design/milestone_1_2_design_doc.md

```diff
@@ -0,0 +1,430 @@
+- email: conj_maintainers@cyberark.com
+  name: Conjur Maintainers
+name: secrets-provider
+version: 1.0.0
+```
+
+Helm Charts are packaged in `tgz` format. 
+
+For each release, we will package our Chart (`helm package secrets-provider`) and the output will be `secrets-provider-<version>.tgz`. This `.tgz` is what we will push to Github.
+
+For more detailed information on how a customer will install our Chart, see the section on [Installation](#installation).
+
+### Milestone 2: Rotation
+
+*TBD*
+
+
+
+### Lifecycle
+
+#### Milestone 1
+
+##### Installation
+
+The customer will be able to override these defaults by creating their own `custom-values.yml` and passing it when installing our Helm chart. For example:
+
+`helm install conjur -f custom-values.yaml https://github.com/cyberark/secrets-provider-for-k8s/helm/secrets-provider-1.0.0.tgz`
+
+Where `custom-values.yaml` is the yaml file that the customer creates to override our defaults and where `https://github.com/cyberark/secrets-provider-for-k8s/helm/secrets-provider-1.0.0.tgz ` is the path to our repository where we will push the packaged Helm Chart.
+
+##### Lifecycle
+
+The lifecycle of the Secrets Provider is independent of the application pods. The application pods that detail K8s Secrets in their environment [will not start](*https://kubernetes.io/docs/concepts/configuration/secret/#details*) until the K8s Secrets they require are populated with the expected key. 
+
+To notify the customer that the Secrets Provider has started and finished it's work, the process will be logged.
+
+##### Update
+
+Because we will be using Helm Charts, we can easily supply customers with the Helm Charts for the current Milestone. All the customer would need to do is delete their current Secrets Provider Chart and install the new one.
+
+It is possible that future Milestones will require additional configurations. If so, the customer can add the necessary parameter with relative easy by manually defining it for us in their `custom-values.yaml` file. If not supplied, we will take our default.
+
+##### Delete
+
+Customers can uninstall the Secrets Provider release by uninstalling the Helm Chart: `helm uninstall secrets-provider `
+
+It is recommended that customers will delete the Helm Chart in its entirety and individual manifests. For example, we recommend that the customer not delete the individual Job manifest itself.
+
+The lifecycle of the Job is the length of the process. In other words, the time it takes to retrieve the Conjur secrets value and update them in the K8s Secrets. To examine the Secrets Provider Job logs, the customer can configure the Job to stay "alive" by adding **ttlSecondsAfterFinished**, with the desired about of seconds, to their Secrets Provider yaml. Because of the nature of a Job, a manual delete is not necessary.
+
+### Order of Deployment
+
+#### Milestone 1
+
+Steps to follow for successful deployment
+
+1. Add all necessary K8s Secrets to `custom-values.yaml`
```

Comment:
Maybe we missed here creation of configmap with secrets to be synced

Reply:
What do you mean?

Reply:
OK, I understood that secrets list  will be provided to helm in config-values.yaml